docs/en/operations/settings/settings.md

```diff
@@ -1727,6 +1727,28 @@ Possible values:
 
 Default value: 0.
 
+## optimize_functions_to_subcolumns {#optimize-functions-to-subcolumns}
+
+Enables or disables optimization by transforming some functions to reading subcolumns. This reduces the amount of data to read. 
+
+These functions can be transformed:
+
+-   [length](../../sql-reference/functions/array-functions.md#array_functions-length) to read the [size0](../../sql-reference/data-types/array.md#array-size) subcolumn.
+-   [empty](../../sql-reference/functions/array-functions.md#function-empty) to read the [size0](../../sql-reference/data-types/array.md#array-size) subcolumn.
```

Comment:
Maybe add expressions to which functions are transformed?

For example:
```empty(arr) -> arr.size0 = 0```

Reply:
Added expressions to function descriptions instead.